### ISSUE TITLE
Create Compose navigation instrumentation

### ISSUE THREAD

**Issue by breedx-splk** [open-telemetry/opentelemetry-android#361]
The existing Activity and Fragment instrumentation via [VisibleScreenTracker](https://github.com/open-telemetry/opentelemetry-android/blob/main/instrumentation/activity/src/main/java/io/opentelemetry/android/instrumentation/activity/VisibleScreenTracker.java) is nice and helpful RUM information. Some apps, however, use (Jetpack) Compose to build sophisticated UI structures, including `@Composable` components that behave (to the user) much like a Fragment or an Activity would.

Compose has a Navigation component that app authors can use to create user flows between these Compose style "screens". This navigation is not yet instrumented, and thus the current screen might appear to be reported incorrectly, if at all. 

Not entirely sure yet what this should look like, but we should build Compose Navigation instrumentation.

**Comment by LikeTheSalad** [comment]
I agree. And I think it would be great if we could come up with a single api to instrument all kinds of screen types, kind of a [VisibleScreenTracker](https://github.com/open-telemetry/opentelemetry-android/blob/main/instrumentation/activity/src/main/java/io/opentelemetry/android/instrumentation/activity/VisibleScreenTracker.java) but for all use cases, Activities, Fragments, and Compose alike.

**Comment by benjoseph-grafana** [comment]
@breedx-splk I'd like to pick this up. Here's a proposed shape, sketching it here for feedback before I open a PR.

### Primary API: an extension on NavController
```
// The canonical entry point — works for ANY controller, however it was obtained.
@Composable
fun NavController.withOpenTelemetry(
    getScreenNameFromDestination: (NavDestination, Bundle?) -> String
): NavController {
    DisposableEffect(this) {
        val navigationListener = NavController.OnDestinationChangedListener { _, dest, args ->
            emitNavigationEvent(
                getScreenNameFromDestination(
                    dest,
                    args
                )
            )
        }
        addOnDestinationChangedListener(navigationListener)
        onDispose { removeOnDestinationChangedListener(navigationListener) }
    }
    return this
}
```
I'd lead with the extension (on the base NavController, so nested/child controllers work too) rather than a rememberNavController wrapper, because a lot of apps don't create the controller inline — it's hoisted, DI-injected, or ViewModel-owned. An extension operates on a reference the developer already holds, so coverage is "can you get the controller?" rather than "did you create it our way?". On top of that I am hoping to add a rememberObservedNavController() convenience that just delegates to the extension.

### What it emits

On each destination change it emits a screen-view event (log record via logsBridge, mirroring compose/click), with the route as app.screen.name.

#### Navigation 3

Nav3 has no NavController/listener - the back stack is developer-owned SnapshotStateList state. It needs a parallel front-end, not the extension. Existing implementations use a side-effect that takes the back stack as a parameter (rather than wrapping its creation). Support for Nav3 can be a separate change.


If the direction looks good I'll open a PR for a new instrumentation/compose/navigation module (sibling of compose/click), with the manual extension + event emitter.
